include/zephyr/sys/util.h

```diff
@@ -250,6 +250,17 @@ extern "C" {
 		((type *)(((char *)(ptr)) - offsetof(type, field))); \
 	})
 
+#ifndef CONCAT
```

Comment:
#ifndef needs to be removed, see https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-3-macro-name-collisions

We still have some using ifndef that predate such rule.